File: psra/repository/src/test/java/by/arhor/psra/repository/AbstractRepositoryTests.java
package by.arhor.psra.repository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

public class AbstractRepositoryTests {
	
	@Autowired protected TagRepository tagRepository;
	@Autowired protected PhotoRepository photoRepository;
	@Autowired protected GalleryRepository galleryRepository;
	@Autowired protected UserRepository userRepository;
	
	@SpringBootApplication
	static class TestApplicationRunner {
		public static void main(String[] args) {
			SpringApplication.run(TestApplicationRunner.class, args);
		}
	}

}

File: psra/repository/src/test/scala/by/arhor/psra/repository/AbstractRepositoryTests.scala
package by.arhor.psra.repository

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

abstract class AbstractRepositoryTests {
	
	@Autowired protected var photoRepository: PhotoRepository = _
	@Autowired protected var galleryRepository: GalleryRepository = _
	@Autowired protected var userRepository: UserRepository = _

}
